### PR TITLE
Improve a few UI elements even more

### PR DESCRIPTION
**Pull request by gabrielvicenteYT**
These include, but are not limited to:
- Text
- Arraylist

### CODE REVIEW

File: src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Text.kt [hunk 1]
```diff
@@ -62,19 +64,19 @@ class Text(x: Double = 10.0, y: Double = 10.0, scale: Float = 1F, side: Side = S
...
+    private val textColor by ListValue("TextColor", arrayOf("Custom", "Rainbow"), "Custom")
+    private val rainbowX by FloatValue("Rainbow-X", -1000F, -2000F..2000F) { textColor == "Rainbow" }
+    private val rainbowY by FloatValue("Rainbow-Y", -1000F, -2000F..2000F) { textColor == "Rainbow" }
 
-    private var red by IntegerValue("Red", 255, 0..255) { !rainbow }
-    private var green by IntegerValue("Green", 255, 0..255) { !rainbow }
-    private var blue by IntegerValue("Blue", 255, 0..255) { !rainbow }
-    private var alpha by IntegerValue("Alpha", 255, 0..255) { !rainbow }
+    private var alpha by IntegerValue("Alpha", 255, 0..255) { textColor == "Custom" }
+    private var red by IntegerValue("Red", 255, 0..255) { alpha > 0 }
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Text.kt]: rgb values don't check if textColor == "Custom"

should preferably be
alphaValue.isSupported() && !alphaValue.isMinimal()

Comment (gabrielvicenteYT) [reply src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Text.kt]: thanks!

**Comment by gabrielvicenteYT** [comment]
@CzechHek The method you showed doesn't work... Anyway, I'm going to change the UI elements even more!

**Comment by gabrielvicenteYT** [comment]
@1zun4 Should I try (and fail) to make something that auto-converts hud.json files that use the old naming system to the new one?

File: src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Arraylist.kt [hunk 2]
```diff
@@ -35,46 +35,43 @@ import java.awt.Color
...
-    private val tags by BoolValue("Tags", true)
-    private val tagsStyle by object : ListValue("TagsStyle", arrayOf("[]", "()", "<>", "-", "|", "Space"), "Space") {
-        override fun isSupported() = tags
+    private val tagsColor by ListValue("Tags-Color", arrayOf("Custom", "Random", "Rainbow"), "Custom")
+    private val tagsAlpha by IntegerValue("Tags-Alpha", 0, 0..255) { tagsColor == "Custom" }
+    private val tagsRed by IntegerValue("Tags-R", 0, 0..255) { tagsColor == "Custom" && tagsAlpha > 0 }
+    private val tagsGreen by IntegerValue("Tags-G", 0, 0..255) { tagsColor == "Custom" && tagsAlpha > 0 }
+    private val tagsBlue by IntegerValue("Tags-B", 0, 0..255) { tagsColor == "Custom" && tagsAlpha > 0 }
+    private val tagsPrefix by TextValue("Tags-Prefix", " §7") { tagsAlpha > 0 }
+    private val tagsSuffix by TextValue("Tags-Suffix", "") { tagsAlpha > 0 }
```
Comment (1zun4) [inline src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Arraylist.kt:70]: text cannot be editied in custom hud

File: src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Arraylist.kt [hunk 3]
```diff
@@ -35,46 +35,43 @@ import java.awt.Color
...
+
+    private fun isColorModeUsed(value: String) = textColor == value || rectMode == value || backgroundColor == value
     private val saturation by FloatValue("Random-Saturation", 0.9f, 0f..1f) { isColorModeUsed("Random") }
     private val brightness by FloatValue("Random-Brightness", 1f, 0f..1f) { isColorModeUsed("Random") }
     private val rainbowX by FloatValue("Rainbow-X", -1000F, -2000F..2000F) { isColorModeUsed("Rainbow") }
     private val rainbowY by FloatValue("Rainbow-Y", -1000F, -2000F..2000F) { isColorModeUsed("Rainbow") }
 
-    private val tags by BoolValue("Tags", true)
-    private val tagsStyle by object : ListValue("TagsStyle", arrayOf("[]", "()", "<>", "-", "|", "Space"), "Space") {
-        override fun isSupported() = tags
```
Comment (1zun4) [inline src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Arraylist.kt:66]: keep this, I like this more.

**Comment by 1zun4** [comment]
I think this PR just adds way too much that is not needed.

**Pull request closed without merging**